File: trip-planner-web/src/api/checklist.ts
import api from './index'

export interface ChecklistData {
  id: number
  tripId: number
  title: string
  category: string
  isChecked: number
  sortOrder: number
}

export interface ChecklistItemReq {
  title: string
  category?: string
}

export const checklistApi = {
  list(tripId: number, category?: string): Promise<ChecklistData[]> {
    return api.get(`/trips/${tripId}/checklist`, { params: { category } })
  },
  create(tripId: number, data: ChecklistItemReq): Promise<ChecklistData> {
    return api.post(`/trips/${tripId}/checklist`, data)
  },
  update(tripId: number, id: number, data: ChecklistItemReq): Promise<ChecklistData> {
    return api.put(`/trips/${tripId}/checklist/${id}`, data)
  },
  delete(tripId: number, id: number): Promise<void> {
    return api.delete(`/trips/${tripId}/checklist/${id}`)
  },
  toggle(tripId: number, id: number): Promise<ChecklistData> {
    return api.patch(`/trips/${tripId}/checklist/${id}/toggle`)
  },
  toggleAll(tripId: number, isChecked: boolean): Promise<ChecklistData[]> {
    return api.patch(`/trips/${tripId}/checklist/toggle-all`, { isChecked })
  },
  loadPreset(tripId: number): Promise<ChecklistData[]> {
    return api.post(`/trips/${tripId}/checklist/preset`)
  }
}

File: trip-planner-flutter/lib/models/checklist.dart
class ChecklistItem {
  final int id;
  final int tripId;
  final String title;
  final String category;
  final bool isChecked;
  final int? sortOrder;

  ChecklistItem({
    required this.id, required this.tripId, required this.title,
    required this.category, required this.isChecked, this.sortOrder,
  });

  factory ChecklistItem.fromJson(Map<String, dynamic> json) => ChecklistItem(
    id: json['id'] as int,
    tripId: json['tripId'] as int,
    title: json['title'] as String,
    category: (json['category'] as String?) ?? 'OTHER',
    isChecked: json['isChecked'] == 1 || json['isChecked'] == true,
    sortOrder: json['sortOrder'] as int?,
  );

  Map<String, dynamic> toJson() => {
    'id': id, 'tripId': tripId, 'title': title,
    'category': category, 'isChecked': isChecked,
    'sortOrder': sortOrder,
  };
}

class ChecklistItemReq {
  final String title;
  final String? category;
  ChecklistItemReq({required this.title, this.category});
  Map<String, dynamic> toJson() => {
    'title': title,
    if (category != null) 'category': category,
  };
}
